optimum/neuron/pipelines/transformers/sentence_transformers.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+
+from transformers.pipelines.base import GenericTensor, Pipeline
+
+from optimum.utils import is_sentence_transformers_available
+
+
+def is_sentence_transformer_model(model: str, token: str = None, revision: str = None):
+    """Checks if the model is a sentence transformer model based on provided model id"""
+    if is_sentence_transformers_available():
+        from optimum.exporters.tasks import TasksManager
+
+        try:
+            _library_name = TasksManager.infer_library_from_model(model, use_auth_token=token, revision=revision)
+            return True if _library_name == "sentence_transformers" else False
+        except ValueError:
+            return False
+    return False
+
+
+class FeatureExtractionPipeline(Pipeline):
+    """
+    Sentence Transformers compatible Feature extraction pipeline uses no model head.
+    This pipeline extracts the sentence embeddings from the sentence transformers, which can be used
+    in embedding-based tasks like clustering and search. The pipeline is based on the `transformers` library.
+    And automatically used instead of the `transformers` library's pipeline when the model is a sentence transformer model.
+
+    Example:
+
+    ```python
+    >>> from optimum.neuron import pipeline
+
+    >>> extractor = pipeline(model="sentence-transformers/all-MiniLM-L6-v2", task="feature-extraction", export=True, batch_size=2, sequence_length=128)
+    >>> result = extractor("This is a simple test.", return_tensors=True)
+    >>> result.shape  # This is a tensor of shape [1, dimension] representing the input string.
+    torch.Size([1, 384])
+    ```
+    """
+
+    def _sanitize_parameters(self, truncation=None, tokenize_kwargs=None, return_tensors=None, **kwargs):
+        if tokenize_kwargs is None:
+            tokenize_kwargs = {}
+
+        if truncation is not None:
+            if "truncation" in tokenize_kwargs:
+                raise ValueError(
+                    "truncation parameter defined twice (given as keyword argument as well as in tokenize_kwargs)"
+                )
+            tokenize_kwargs["truncation"] = truncation
+
+        preprocess_params = tokenize_kwargs
+
+        postprocess_params = {}
+        if return_tensors is not None:
+            postprocess_params["return_tensors"] = return_tensors
+
+        return preprocess_params, {}, postprocess_params
+
+    def preprocess(self, inputs, **tokenize_kwargs) -> Dict[str, GenericTensor]:
```

Comment:
My primary concern at this time is that the Sentence Transformer tokenizer uses [this `max_seq_length`](https://huggingface.co/sentence-transformers/all-mpnet-base-v2/blob/main/sentence_bert_config.json#L2) as the "correct" maximum length as opposed to the value defined in the `tokenizer_config.json`. 

Here, we are relying on the tokenizer defined in the `Pipeline`, which won't use the max_seq_length. As a result, I think this ST pipeline component will perform differently (worse, to be precise) for longer input texts. A solution is to use `model_inputs = self.model.tokenize(inputs)` instead. 
Do note that the ST tokenize method does not allow for extra tokenize kwargs such as truncation, return_tensors, or padding. These are unfortunately hardcoded at the moment.

Reply:
Yet another solution is to rely exclusively on `self.model.encode(...)` in `def _forward`, but I recognize that this might clash with some requirements of the `Pipeline`.

Reply:
@tomaarsen for inferentia models need to be traced to a sequence length before running inference since we have static shapes. You always need to specify a `sequence_length` and `batch_size` before you can compile a model which is then used. 
This abstracted away by the user in the NeuronModelForSentenceTransformers class.

Reply:
There is no "sentence-transformers" used at all at the end. since the model is traced and it happens with transformers. We should be good here.

Reply:
See here: https://github.com/huggingface/optimum-neuron/blob/18460aa22407586e82b345745a9fc55d9f4828b3/optimum/neuron/modeling.py#L201

Reply:
Ahh, I see! Thanks for the heads up. I figured it was more like Intel Gaudi, which does just work with regular Sentence Transformers (as long as the padding is "max_length" to also get static shapes & the device is "hpu"). 

Then my concern still stands: I think the [`max_seq_length`](https://huggingface.co/sentence-transformers/all-mpnet-base-v2/blob/main/sentence_bert_config.json#L2) might not be taken into account correctly.

Reply:
fyi, `max_seq_length` is not taken into account at all by the export of Neuron model. Shall we prevent users from setting static seq len higher than this value?

Reply:
For Sentence Transformer models, `max_seq_length` should have priority. In 99% of cases, this is stored in the `sentence_bert_config.json` file. You might indeed want to store it in a `config.json` when exporting, or override `model_max_length` in `tokenizer_config.json` as that should work in a more "expected" fashion.